### PR TITLE
[SQL] [WIP] Blacklists several Hive 0.13.1 specific test cases for Hive 0.12.0

### PR DESCRIPTION
**Pull request by liancheng**
[Review on Reviewable](https://reviewable.io/reviews/apache/spark/4851)

### CODE REVIEW

File: sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala [hunk 1]
```diff
@@ -220,7 +220,18 @@ private[hive] object HiveShim {
     "serde_regex",
     "udf_to_date",
     "udaf_collect_set",
-    "udf_concat"
+    "udf_concat",
+
+    // The following 3 cases use 0.13.1 decimal syntax
+    "udf_to_double",
+    "udf_to_float",
+    "udf_pmod",
```
Comment (adrian-wang) [inline sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala]: These 3 cases also use different ".q" files.

**Comment by adrian-wang** [comment]
so, what's the relationship between this and #4107 ?

**Comment by liancheng** [comment]
The Hive shim layer shirinking part in #4107 is not that essential, but hard to review and touches lots of files. However, the Hive 0.12.0 tests part is essential. This PR extracts the latter from #4107.

**Comment by marmbrus** [comment]
ping.  do we need this if we are going to be getting rid of the ability to run multiple versions of hive execution and only support multiple metastore versions?

**Comment by liancheng** [comment]
No. With the metastore adapter layer, we can always keep our tests consistent with the most recent Hive version.

**Comment by marmbrus** [comment]
Okay, lets close this issue then.  I think we are on track to do the metadata / execution split for hive in 1.4

**Pull request closed without merging**